### ISSUE TITLE
[no-useless-constructor] False-positive if superclass constructor is protected

### ISSUE THREAD

**Issue by cowwoc** [typescript-eslint/typescript-eslint#2690]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
	"no-useless-constructor": "off",
	"@typescript-eslint/no-useless-constructor": "error",
  }
}
```

```TS
class Parent {
  protected constructor() {
  }
}

class Child extends Parent {
  constructor() {
  }
}
```

tsconfig.json:
```
{
	// Based on https://stackoverflow.com/a/61305579/14731 and @tsconfig/node14/tsconfig.json
	"compilerOptions": {
		"preserveConstEnums": true,
		"allowSyntheticDefaultImports": true,
		"target": "ES2020",
		"lib": [
			"ES2020",
			"DOM"
		],
		"module": "ES2020",
		"moduleResolution": "node",
		"strict": true,
		"esModuleInterop": true,
		"skipLibCheck": true,
		"forceConsistentCasingInFileNames": true
	},
	"declaration": true
}
```

**Expected Result**

Rule shouldn't complain about child's constructor.

**Actual Result**

Rule claims child constructor is useless, but if it is removed I get a compiler error trying to instantiate a `new Child()`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.4.1` |
| `@typescript-eslint/parser`        | `4.4.1` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.11.0` |
| `node`                             | `14.11.0` |
###

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/948
